### ISSUE TITLE
Should type of TorrentStatus.state be TorrentState instead of string?

### ISSUE THREAD

**Issue by jerblack** [gdm85/go-libdeluge#37]
This may be a dumb question, but I saw that you went to the trouble of defining a TorrentState type and constants for each of the possible torrent states, so I tried to do an == comparison but couldn't because TorrentStatus.state is a string and the state constants are type TorrentState. Shouldn't TorrentStatus.state be type TorrenState?
```golang
// this should work but doesn't
tors, _ := d.client.TorrentsStatus(delugeclient.StateUnspecified, nil)
if tors[0].State == delugeclient.StateSeeding {
  // seeding!
}
```
This works if I cast the TorrentState constant to a string ( `string(delugeclient.StateSeeding)` ) or cast the torrent state from string to TorrentState ( `delugeclient.TorrentState(tors[0].State)` ) but it seems like they should already be the same type.

**Comment by gdm85** [comment]
No, it is not a dumb question - I think it does not have the right type because of the "conversion magic" that happens when decoding the python RPC data.

I am not sure how to easily fix this at the moment :thinking:

**Comment by Krimlocke** [comment]
Wouldn't this be as simple as changing `State string` to `State TorrentState` in the `TorrentStatus` struct? Since `TorrentState` is defined as `type TorrentState string`, the rencode decoder should handle the assignment from a raw string without any changes to the decoding logic. If not, it's a one-line fix in the decoder too.
Happy to open a PR for this if you'd like? I'm already looking at the library for another small addition (adding the `trackers` field to `TorrentStatus`, which I opened an issue for as well).

**Comment by gdm85** [comment]
My skepticism comes from how the rencode library would need to handle this: either a reflection fallback or completely switching to reflection instead of type switches.

However it would also make the convertAssign broader and bugs might be introduced this way.

Can't your problem be solved by using a cast?

I am inclined to keep the object layer "dumb" so that it can always be mapped to scalar types and more easily verify it.
